__tests__/__unit__/uss/ZoweUSSNode.unit.test.ts

```diff
@@ -0,0 +1,617 @@
+
+    Object.defineProperty(Profiles, "getInstance", {
+        value: jest.fn(() => {
+            return {
+                allProfiles: [{name: "firstName"}, {name: "secondName"}],
+                defaultProfile: {name: "firstName"},
+                getDefaultProfile: mockLoadNamedProfile,
+                promptCredentials: jest.fn(()=> {
+                    return ["fake", "fake", "fake"];
+                }),
+                loadNamedProfile: mockLoadNamedProfile,
+                usesSecurity: true,
+                validProfile: ValidProfileEnum.VALID,
+                checkCurrentProfile: jest.fn(),
+                getProfiles: jest.fn(() => {
+                    return [{name: profileOne.name, profile: profileOne}, {name: profileOne.name, profile: profileOne}];
+                }),
+                refresh: jest.fn(),
+            };
+        })
+    });
+
+    beforeEach(() => {
+        ussFile.mockReset();
+        openTextDocument.mockReset();
+        showTextDocument.mockReset();
+        showErrorMessage.mockReset();
+        existsSync.mockReset();
+        showQuickPick.mockReset();
+        showInputBox.mockReset();
+
+        ussFile.mockReturnValue(response);
+        withProgress.mockReturnValue(response);
+        openTextDocument.mockResolvedValue("test.doc");
+        showInputBox.mockReturnValue("fake");
+
+        ussNode = new ZoweUSSNode("usstest", vscode.TreeItemCollapsibleState.Expanded, null, session, null, null, profileOne.name, "123");
+        dsNode = new ZoweUSSNode("testSess", vscode.TreeItemCollapsibleState.Expanded, ussNode, generateISessionWithoutCredentials(), null);
+    });
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it("Tests that node.openUSS() is executed successfully", async () => {
+        const node = new ZoweUSSNode("node", vscode.TreeItemCollapsibleState.None, ussNode, session, "/", false, profileOne.name);
+
+        const isBinSpy = jest.spyOn(ussApi, "isFileTagBinOrAscii");
+        existsSync.mockReturnValue(null);
+
+        // Tests that correct file is downloaded
+        await node.openUSS(false, true, testUSSTree);
+        expect(existsSync.mock.calls.length).toBe(1);
+        expect(existsSync.mock.calls[0][0]).toBe(path.join(USS_DIR, "/" + node.mProfileName + "/", node.fullPath));
+        expect(isFileTagBinOrAscii.mock.calls.length).toBe(1);
+        expect(isFileTagBinOrAscii.mock.calls[0][0]).toBe(session);
+        expect(isFileTagBinOrAscii.mock.calls[0][1]).toBe(node.fullPath);
+        expect(withProgress).toBeCalledWith(
+            {
+                location: vscode.ProgressLocation.Notification,
+                title: "Opening USS file..."
+            }, expect.any(Function)
+        );
+
+        // Tests that correct file is opened in editor
+        withProgress(downloadUSSFile);
+        expect(withProgress).toBeCalledWith(downloadUSSFile);
+        expect(openTextDocument.mock.calls.length).toBe(1);
```

Comment:
I thought mock.calls.length should not be tested or part of the expect? Or is that just a recommendation?

Reply:
I didn't change the functions, as much as I could avoid it. I just moved them into the correct file.

If some functionality can be tested another way, I believe the goal is to eliminate expect statements like this one...